opentelemetry-sdk/src/opentelemetry/sdk/_metrics/_internal/metric_reader.py

```diff
@@ -162,7 +217,12 @@ def _set_collect_callback(
         self._collect = func
 
     @abstractmethod
-    def _receive_metrics(self, metrics: Iterable[Metric]):
+    def _receive_metrics(
+        self,
+        metrics: Dict[
```

Comment:
re using dict here. What do you think of using a type annotation compatible with generators to keep the door open to streaming stuff through? If we make the intended usage

```py
for resource, instrumentation_scope_metrics in metrics:
  for instrumentation_scope, metrics:
    # ...
```

I reckon it's not important to key into the dictionary for exporters, and folks generally want to iterate over. OTOH, maybe there are bigger memory savings possible by caching some objects. But we could leave the door open.

Reply:
If you do want to stick with Dict, I would recommend `Mapping` which is immutable interface instead

Reply:
All right, using `Mapping` instead :+1: